### PR TITLE
Implement Java modules

### PR DESCRIPTION
**Pull request by plata**

### CODE REVIEW

**Comment by madoar** [comment]
@plata shall I still take a look at `PhoenicisFilteredList`?

**Comment by plata** [comment]
What's left for this PR is to fix Travis. To be honest, I don't get the Java 11 issue.

File: phoenicis-javafx/src/test/java/org/phoenicis/javafx/AppConfigurationInitialisationTest.java [hunk 1]
```diff
@@ -16,8 +16,17 @@
  */
 public class AppConfigurationInitialisationTest {
 
+    public static void setupHeadless() {
```
Comment (madoar) [inline phoenicis-javafx/src/test/java/org/phoenicis/javafx/AppConfigurationInitialisationTest.java]: Please add some JavaDoc explaining what the method is responsible for and what the different options you set do.

**Comment by madoar** [comment]
Just asking out of interest:
Did you somehow generate a first version of the module-info files which you could then edit manually or did you create them completely by hand?

**Comment by plata** [comment]
I created the files and then used IntelliJ to fill them. Here's the tutorial: https://dzone.com/articles/java-9-modules-part-2-intellij-and-maven

File: phoenicis-javafx/src/test/java/org/phoenicis/javafx/AppConfigurationInitialisationTest.java (hunk 1 above)
Comment (plata) [reply phoenicis-javafx/src/test/java/org/phoenicis/javafx/AppConfigurationInitialisationTest.java]: Done.

**Comment by plata** [comment]
@madoar can you check the Oracle JDK 8 build? I think you have Oracle JDK, right?

**Comment by plata** [comment]
Might be. Can you test? For me it's a bit difficult because I do not have the error with OpenJDK so I would have to test every change in Travis.

**Comment by madoar** [comment]
Adding:
```xml
            <build>
                <pluginManagement>
                    <plugins>
                        <plugin>
                            <groupId>org.apache.maven.plugins</groupId>
                            <artifactId>maven-surefire-plugin</artifactId>
                            <version>3.0.0-M1</version>
                            <configuration>
                                <forkCount>0</forkCount>
                                <excludes>
                                    <exclude>module-info.java</exclude>
                                </excludes>
                            </configuration>
                        </plugin>
                    </plugins>
                </pluginManagement>
            </build>
```
to the `jdk8` profile in `phoenicis-javafx/pom.xml` doesn't seem to help.

**Comment by madoar** [comment]
Yes and it only occurs during the testing phase, because `mvn clean package -DskipTests` results in a `BUILD SUCCESS`

**Comment by plata** [comment]
So maybe this is solved automatically once we merge your other PR.

**Comment by plata** [comment]
@madoar the problem is: The required classes are part of JavaFX (javafx-base). I cannot use the OpenJFX dependencies because then we have again the issue that it was built with a newer Java version. It should use the classes from the JFX included in the JDK but for some reason it doesn't.

**Comment by plata** [comment]
Builds now but I cannot run it.

**Comment by madoar** [comment]
What do you mean with I can not run it?
Does it not start or do you require Oracle JDK?

**Comment by plata** [comment]
It doesn't start (I get split package errors). That's nothing related to OracleJDK though.

**Comment by plata** [comment]
I think I got a bit further. Right now, I'm having issues opening the resource bundles for the translations.

**Comment by plata** [comment]
This seems to be working now. I ran with Java 8 and 10 and also the flatpak. @ImperatorS79 can you check that the deb is still working?

**Comment by madoar** [comment]
This is interesting.
When running `mvn clean package` on this PR, I get the following result output:

```
...
[INFO] Reactor Summary:
[INFO] 
[INFO] phoenicis-configuration ............................ SUCCESS [  6.453 s]
[INFO] phoenicis-entities ................................. SUCCESS [  0.176 s]
[INFO] phoenicis-multithreading ........................... SUCCESS [  0.540 s]
[INFO] phoenicis-win32 .................................... SUCCESS [  0.436 s]
[INFO] phoenicis-tools .................................... SUCCESS [  6.914 s]
[INFO] phoenicis-repository ............................... SUCCESS [  1.011 s]
[INFO] phoenicis-scripts .................................. SUCCESS [  0.169 s]
[INFO] phoenicis-library .................................. SUCCESS [  0.133 s]
[INFO] phoenicis-engines .................................. SUCCESS [  0.122 s]
[INFO] phoenicis-settings ................................. SUCCESS [  0.067 s]
[INFO] phoenicis-containers ............................... SUCCESS [  0.115 s]
[INFO] phoenicis-javafx ................................... SUCCESS [  4.049 s]
[INFO] phoenicis-cli ...................................... SUCCESS [  1.208 s]
[INFO] phoenicis-dist ..................................... SUCCESS [  1.855 s]
[INFO] phoenicis-tests .................................... SUCCESS [  1.019 s]
[INFO] phoenicis .......................................... SUCCESS [  0.014 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 24.471 s
[INFO] Finished at: 2018-11-25T12:57:06+01:00
[INFO] Final Memory: 204M/1175M
[INFO] ------------------------------------------------------------------------
--- Exception in Async Thread ---
java.lang.AssertionError: Internal inconsistency - wrong EventLoop
        com.sun.glass.ui.EventLoop.enter(EventLoop.java:108)
        com.sun.javafx.tk.quantum.QuantumToolkit.enterNestedEventLoop(QuantumToolkit.java:590)
        javafx.stage.Stage.showAndWait(Stage.java:474)
        javafx.scene.control.HeavyweightDialog.showAndWait(HeavyweightDialog.java:162)
        javafx.scene.control.Dialog.showAndWait(Dialog.java:341)
        org.phoenicis.javafx.views.common.ErrorMessage.<init>(ErrorMessage.java:108)
        org.phoenicis.javafx.views.common.ErrorMessage.<init>(ErrorMessage.java:60)
        org.phoenicis.javafx.controller.containers.ContainersController.lambda$null$2(ContainersController.java:71)
        com.sun.javafx.application.PlatformImpl.lambda$null$172(PlatformImpl.java:295)
        java.security.AccessController.doPrivileged(Native Method)
        com.sun.javafx.application.PlatformImpl.lambda$runLater$173(PlatformImpl.java:294)
        com.sun.glass.ui.monocle.RunnableProcessor.runLoop(RunnableProcessor.java:92)
        com.sun.glass.ui.monocle.RunnableProcessor.enterNestedEventLoop(RunnableProcessor.java:107)
        com.sun.glass.ui.monocle.MonocleApplication._enterNestedEventLoop(MonocleApplication.java:144)
        com.sun.glass.ui.Application.enterNestedEventLoop(Application.java:511)
        com.sun.glass.ui.EventLoop.enter(EventLoop.java:107)
        com.sun.javafx.tk.quantum.QuantumToolkit.enterNestedEventLoop(QuantumToolkit.java:590)
        javafx.stage.Stage.showAndWait(Stage.java:474)
        javafx.scene.control.HeavyweightDialog.showAndWait(HeavyweightDialog.java:162)
        javafx.scene.control.Dialog.showAndWait(Dialog.java:341)
        org.phoenicis.javafx.views.common.ErrorMessage.<init>(ErrorMessage.java:108)
        org.phoenicis.javafx.views.common.ErrorMessage.<init>(ErrorMessage.java:60)
        org.phoenicis.javafx.controller.containers.ContainersController.lambda$null$0(ContainersController.java:66)
        com.sun.javafx.application.PlatformImpl.lambda$null$172(PlatformImpl.java:295)
        java.security.AccessController.doPrivileged(Native Method)
        com.sun.javafx.application.PlatformImpl.lambda$runLater$173(PlatformImpl.java:294)
        com.sun.glass.ui.monocle.RunnableProcessor.runLoop(RunnableProcessor.java:92)
        com.sun.glass.ui.monocle.RunnableProcessor.run(RunnableProcessor.java:51)
        java.lang.Thread.run(Thread.java:748)

--- Exception in Async Thread ---
java.lang.AssertionError: Internal inconsistency - wrong EventLoop
        com.sun.glass.ui.EventLoop.enter(EventLoop.java:108)
        com.sun.javafx.tk.quantum.QuantumToolkit.enterNestedEventLoop(QuantumToolkit.java:590)
        javafx.stage.Stage.showAndWait(Stage.java:474)
        javafx.scene.control.HeavyweightDialog.showAndWait(HeavyweightDialog.java:162)
        javafx.scene.control.Dialog.showAndWait(Dialog.java:341)
        org.phoenicis.javafx.views.common.ErrorMessage.<init>(ErrorMessage.java:108)
        org.phoenicis.javafx.views.common.ErrorMessage.<init>(ErrorMessage.java:60)
        org.phoenicis.javafx.controller.containers.ContainersController.lambda$null$0(ContainersController.java:66)
        com.sun.javafx.application.PlatformImpl.lambda$null$172(PlatformImpl.java:295)
        java.security.AccessController.doPrivileged(Native Method)
        com.sun.javafx.application.PlatformImpl.lambda$runLater$173(PlatformImpl.java:294)
        com.sun.glass.ui.monocle.RunnableProcessor.runLoop(RunnableProcessor.java:92)
        com.sun.glass.ui.monocle.RunnableProcessor.run(RunnableProcessor.java:51)
        java.lang.Thread.run(Thread.java:748)

```

**Comment by plata** [comment]
I have seen this as well when building with JDK 8. No idea where it's coming from. Also, it doesn't seem to affect the build in any way. Is it causing issues for you?

**Comment by plata** [comment]
Maybe it's not necessary. Let me check something.

**Comment by plata** [comment]
@madoar can you try again?

**Comment by madoar** [comment]
Building works now for me.

**Comment by ImperatorS79** [comment]
I have got:
```
Error occurred during initialization of boot layer
java.lang.module.ResolutionException: Modules xmlParserAPIs and java.xml export package org.xml.sax.ext to module org.apache.commons.compress
```
When trying to launch phoenicis-javafx from .deb.

**Comment by plata** [comment]
@ImperatorS79 that's strange. I'm excluding xmlParserAPIs from the lib folder to avoid exactly this. Do you maybe have old .jars left over from a previous installation?

**Pull request merged**